torch/testing/_internal/common_device_type.py

```diff
@@ -756,6 +756,8 @@ def _parametrize_test(self, test, generic_cls, device_cls):
                     if dtype in dtype_set:
                         dtypes = {dtype}
                         break
+                else:
+                    dtypes = {}
```

Comment:
This preserves the previous behaviour (rather than skipping the test if this happens. But yeah, this could be perhaps an assert(false). We can do it in a follow-up PR